### PR TITLE
Fix S3 local test

### PR DESCRIPTION
**Pull request by CoderHam**
Disable polling
- Use model control instead to workaround Minio timestamp issue.
- Fix build to work when only S3 is enabled and GCS is disabled

### CODE REVIEW

File: qa/L0_s3_local/test.sh [hunk 1]
```diff
@@ -71,15 +71,19 @@ export AWS_ACCESS_KEY_ID=minio && \
...
+    awslocal --endpoint-url=http://localhost:4572 s3 rm s3://demo-bucket && \
+    awslocal s3 rb s3://demo-bucket && \
+    awslocal s3 mb s3://demo-bucket && \
     awslocal s3 sync $DATADIR s3://demo-bucket
 
 RET=0
 
 # Test with hostname
-SERVER_ARGS="--log-verbose=1 --model-repository=s3://localhost:4572/demo-bucket"
+SERVER_ARGS="--model-repository=s3://localhost:4572/demo-bucket --allow-poll-model-repository=false --allow-model-control=true"
```
Comment (deadeyegoodwin) [inline qa/L0_s3_local/test.sh:82]: --model-control-mode=none is the new way to turn off polling. The flags you are using will be removed in Triton V2.

File: qa/L0_s3_local/test.sh [hunk 2]
```diff
@@ -90,24 +94,28 @@ if [ "$SERVER_PID" == "0" ]; then
...
-for MODEL_NAME in resnet_v1_50_graphdef resnet_v1_50_savedmodel; do
-  $PERF_CLIENT -m $MODEL_NAME -p 3000 -t 1 >$CLIENT_LOG 2>&1
-  if [ $? -ne 0 ]; then
-      echo -e "\n***\n*** Test Failed\n***"
-      cat $CLIENT_LOG
-      RET=1
-  fi
+for MODEL_NAME in $MODEL_NAMES; do
+    code=`curl -s -w %{http_code} -X POST localhost:8000/api/modelcontrol/load/${MODEL_NAME}_float32_float32_float32`
+    if [ "$code" == "400" ]; then
```
Comment (deadeyegoodwin) [inline qa/L0_s3_local/test.sh:100]: Test for anything non-200 as a failure.

File: qa/L0_s3_local/test.sh [hunk 3]
```diff
@@ -90,24 +94,28 @@ if [ "$SERVER_PID" == "0" ]; then
...
+    fi
 done
 set -e
 
 kill $SERVER_PID
 wait $SERVER_PID
 
 # Test with IP
-SERVER_ARGS="--log-verbose=1 --model-repository=s3://127.0.0.1:4572/demo-bucket"
+SERVER_ARGS="--model-repository=s3://127.0.0.1:4572/demo-bucket --allow-poll-model-repository=false --allow-model-control=true"
```
Comment (deadeyegoodwin) [inline qa/L0_s3_local/test.sh:118]: same as above

File: qa/L0_s3_local/test.sh [hunk 4]
```diff
@@ -120,16 +128,20 @@ if [ "$SERVER_PID" == "0" ]; then
...
-for MODEL_NAME in resnet_v1_50_graphdef resnet_v1_50_savedmodel; do
-  $PERF_CLIENT -m $MODEL_NAME -p 3000 -t 1 >$CLIENT_LOG 2>&1
-  if [ $? -ne 0 ]; then
-      echo -e "\n***\n*** Test Failed\n***"
-      cat $CLIENT_LOG
-      RET=1
-  fi
+for MODEL_NAME in $MODEL_NAMES; do
+    code=`curl -s -w %{http_code} -X POST localhost:8000/api/modelcontrol/load/${MODEL_NAME}_float32_float32_float32`
+    if [ "$code" == "400" ]; then
```
Comment (deadeyegoodwin) [inline qa/L0_s3_local/test.sh:134]: fix

File: qa/L0_s3_local/test.sh [hunk 5]
```diff
@@ -71,15 +71,19 @@ export AWS_ACCESS_KEY_ID=minio && \
...
     awslocal s3 sync $DATADIR s3://demo-bucket
 
 RET=0
 
 # Test with hostname
-SERVER_ARGS="--log-verbose=1 --model-repository=s3://localhost:4572/demo-bucket"
+SERVER_ARGS="--model-repository=s3://localhost:4572/demo-bucket --allow-poll-model-repository=false --allow-model-control=true"
 SERVER_LOG="./inference_server_hostname.log"
 
+MODEL_NAMES="graphdef libtorch netdef onnx plan savedmodel"
```
Comment (GuanLuo) [inline qa/L0_s3_local/test.sh:85]: change to `BACKENDS`, not yet a model name

**Pull request merged**